### PR TITLE
Feat/add refund function

### PR DESCRIPTION
**Pull request by TheYakuza**
add functions to refund amount at patpass complete transaction

### CODE REVIEW

File: lib/transaccion_completa/Transaction.php [hunk 1]
```diff
@@ -206,8 +208,114 @@ public static function commit(
         $transactionCommitResponse = new TransactionCommitResponse($responseJson);
 
         return $transactionCommitResponse;
+    }
+
+    public function refund(
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/Transaction.php]: All of these methods should be static

Comment (TheYakuza) [reply lib/transaccion_completa/Transaction.php]: fixed at 334bdf4

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php [hunk 2]
```diff
@@ -14,5 +14,262 @@
...
+        $this->setVci($vci);
+        $amount = isset($json["amount"]) ? $json["amount"] : null;
+        $this->setAmount($amount);
+        $status = isset($json["status"]) ? $json["status"] : null;
+        $this->setStatus($status);
+        $buyOrder = isset($json["buy_order"]) ? $json["buy_order"] : null;
+        $this->setBuyOrder($buyOrder);
+        $sessionId = isset($json["session_id"]) ? $json["session_id"] : null;
+        $this->setSessionId($sessionId);
+        $cardDetail= isset($json["accounting_date"]) ? $json["accounting_date"] : null;
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionStatusResponse.php]: this is  wrong

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php [hunk 3]
```diff
@@ -14,5 +14,262 @@
...
+        $this->setCardDetail($cardDetail);
+        $accountingDate = isset($json["card_detail"]) ? $json["card_detail"] : null;
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionStatusResponse.php]: This is also wrong

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php [hunk 4]
```diff
@@ -14,5 +14,262 @@
 
 class TransactionStatusResponse
 {
+    public $vci;
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionStatusResponse.php]: Is there no installments amount here?

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php [hunk 5]
```diff
@@ -14,5 +14,262 @@
...
+    public $status;
+    public $buyOrder;
+    public $sessionId;
+    public $cardDetail; // card_number
+    public $accountingDate;
+    public $transactionDate;
+    public $authorizationCode;
+    public $paymentTypeCode;
+    public $responseCode;
+    public $installmentsNumber;
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionStatusResponse.php]: Does this operation not have `installments_amount`

File: lib/transaccion_completa/transaction/TransactionRefundResponse.php [hunk 6]
```diff
@@ -14,5 +14,138 @@
 
 class TransactionRefundResponse
 {
+    public $type;
+    public $authorizationCode;
+    public $authorizacionDate;
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionRefundResponse.php]: `authorizacionDate` should be `authorizationDate`

File: lib/transaccion_completa/transaction/TransactionRefundResponse.php [hunk 7]
```diff
@@ -14,5 +14,138 @@
...
+    public function getAuthorizacionDate()
+    {
+        return $this->AuthorizacionDate;
+    }
+
+    /**
+     * @param mixed $AuthorizacionDate
+     * @return TransactionRefundResponse
+     */
+    public function setAuthorizacionDate($authorizacionDate)
```
Comment (eugenioLopezRamos) [inline lib/transaccion_completa/transaction/TransactionRefundResponse.php]: Same, fix the typo

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 5 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: In the docs doesn't have that
image

File: lib/transaccion_completa/Transaction.php [hunk 8]
```diff
@@ -16,16 +16,18 @@
...
 
 class Transaction
 {
     const CREATE_TRANSACTION_ENDPOINT  = '/rswebpaytransaction/api/webpay/v1.0/transactions';
     const INSTALLMENTS_TRANSACTION_ENDPOINT = '/rswebpaytransaction/api/webpay/v1.0/transactions/$TOKEN$/installments';
     const COMMIT_TRANSACTION_ENDPOINT = '/rswebpaytransaction/api/webpay/v1.0/transactions/$TOKEN$';
-    const REFUND_TRANSACTION_ENDPOINT = '';
-    const STATUS_TRANSACTION_ENDPOINT = '';
+    const REFUND_TRANSACTION_ENDPOINT = '/rswebpaytransaction/api/webpay/v1.0/transactions/$TOKEN$/refunds';
+    const STATUS_TRANSACTION_ENDPOINT = ' /rswebpaytransaction/api/webpay/v1.0/transactions/$TOKEN$';
```
[the remark this answers is not in the document]
Comment (TheYakuza) [reply lib/transaccion_completa/Transaction.php]: typo, fixed at 0593d63

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 3 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: same response in https://github.com/TransbankDevelopers/transbank-sdk-php/pull/96#discussion_r311258300

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 5 above)
Comment (eugenioLopezRamos) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: Fair enough. Do check what the API responds

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 3 above)
Comment (eugenioLopezRamos) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: ??? I don't understand the answer. You are assigning the `card_detail` to `accountingDate`

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 2 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: fixed on a4f69db

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 3 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: fixed on a4f69db

File: lib/transaccion_completa/transaction/TransactionStatusResponse.php (hunk 5 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionStatusResponse.php]: I've checked the response using postman and you're right, installments amount is in the response, so now I've added to it fixed on a4f69db

File: lib/transaccion_completa/transaction/TransactionRefundResponse.php (hunk 6 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionRefundResponse.php]: fixed typo on 85ed5da

File: lib/transaccion_completa/transaction/TransactionRefundResponse.php (hunk 7 above)
Comment (TheYakuza) [reply lib/transaccion_completa/transaction/TransactionRefundResponse.php]: fixed typo on 85ed5da

**Comment by TheYakuza** [comment]
fix merge conflict in composer.json

**Comment by TheYakuza** [comment]
@eugenioLopezRamos can you please check again? I've used this branch to test also the test-project and not problems seen at this time

**Pull request merged**